core/src/execution/datafusion/planner.rs

```diff
@@ -1397,7 +1397,18 @@ impl PhysicalPlanner {
             args.is_empty(),
         ));
 
-        Ok(scalar_expr)
+        match fun_name.as_str() {
```

Comment:
Hmm, please don't put this special handling in `create_scalar_function_expr`. Could you move it to `create_comet_physical_fun` where special handling of scalar functions should be located?

Reply:
Thanks for bringing that up!

Just to clarify, if we were to do this as part of `create_comet_physical_fun` then it would be all reimplemented at a lower level and not able to reuse the Datafusion Log2 implementation via an IfExpr.  Is that right? 

Happy to go either way. @andygrove's original hint mentioned using IfExpr, but I couldn't quite get that working inside of `create_comet_physical_fun` because it doesn't have access to the arg exprs.

Reply:
Apologies for the delay in responding here. I will start reviewing this on Monday.